components/server/src/workspace/workspace-starter.ts

```diff
@@ -426,14 +426,32 @@ export class WorkspaceStarter {
         reason: string,
         policy?: StopWorkspacePolicy,
     ): Promise<void> {
-        ctx.span?.setTag("stopWorkspaceReason", reason);
+        const span = TraceContext.startSpan("stopWorkspaceInstance", ctx);
+        span.setTag("stopWorkspaceReason", reason);
         log.info({ instanceId }, "Stopping workspace instance", { reason });
 
         const req = new StopWorkspaceRequest();
         req.setId(instanceId);
         req.setPolicy(policy || StopWorkspacePolicy.NORMALLY);
 
         const client = await this.clientProvider.get(instanceRegion);
+        if (!client) {
```

Comment:
How do we know that it's not a temporary scenario and the cluster is indeed actually decommissioned?